testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -766,6 +766,63 @@ impl MultipleMinerTest {
         Ok(txid)
     }
 
+    pub fn send_contract_publish(&mut self, contract_name: &str, contract_src: &str) -> String {
+        let http_origin = format!(
+            "http://{}",
+            &self.signer_test.running_nodes.conf.node.rpc_bind
+        );
+        let contract_tx = make_contract_publish(
+            &self.sender_sk,
+            self.sender_nonce,
+            self.send_fee,
+            self.signer_test.running_nodes.conf.burnchain.chain_id,
+            contract_name,
+            contract_src,
+        );
+        self.sender_nonce += 1;
+        submit_tx(&http_origin, &contract_tx)
+    }
+
+    /// Sends a contract publish tx to the stacks node and waits for the stacks node to mine it
+    /// Returns the txid of the transfer tx.
+    pub fn send_and_mine_contract_publish(
+        &mut self,
+        contract_name: &str,
+        contract_src: &str,
+        timeout_secs: u64,
+    ) -> Result<String, String> {
+        let stacks_height_before = self.get_peer_stacks_tip_height();
+        let txid = self.send_contract_publish(contract_name, contract_src);
+        wait_for(timeout_secs, || {
+            Ok(self.get_peer_stacks_tip_height() > stacks_height_before)
+        })?;
+        Ok(txid)
+    }
+
+    pub fn send_contract_call(
+        &mut self,
+        contract_name: &str,
+        function_name: &str,
+        function_args: &[clarity::vm::Value],
+    ) -> String {
+        let http_origin = format!(
+            "http://{}",
+            &self.signer_test.running_nodes.conf.node.rpc_bind
+        );
+        let contract_tx = make_contract_call(
+            &self.sender_sk,
+            self.sender_nonce,
+            self.send_fee,
```

Comment:
Again, we might have a problem with the fee here.

Reply:
I don't think this changed at all since my comment, did it? The simplest fix here might be to just build the transaction with some default fee, then build it again, setting the fee to the sized of the generated transaction.

Reply:
It seems to work for this case, because the size is 179, but it would not work generically if this function is used in other cases.

Reply:
oops, i did it only for contract publish not call. fixing it...